### PR TITLE
[5.5] [Concept] PSR-11 compliant container

### PR DESCRIPTION
**Pull request by deleugpn**
Following up on laravel/internals#550 I drafted up this commit to draw traction to the debate of whether Laravel container should be PSR-11 compliant or not. 

Given that Laravel container already offers a complete set of features, the compliance would probably warrant just the exposure of `has` and `get` method. I'm not convinced that the benefits of being compliant out-weight the necessity of maintaining the dependency. 

I know it may sound weird to open a PR and discredit it, but like I said my goal is to bring traction to the subject now that PSR-11 was accepted.

### CODE REVIEW

**Comment by thecrypticace** [comment]
Does the spec explicitly prevent `has()` from returning true for autowired objects? Ex: I didn't bind an object to the container but it can still be built by it.

If so then this implementation looks fine to me. If not, I wonder if it might be worth checking to see if an object is buildable (idk try to create it?) by catching the `BidingResolutionException` and turning it into the `EntryNotFoundException` if need be. Not sure how much of a perf. hit that'd be though.

File: src/Illuminate/Container/Container.php [hunk 1]
```diff
@@ -1222,4 +1222,24 @@ public function __set($key, $value)
...
+
+        throw new EntryNotFoundException();
+    }
+
+    /**
+     *  {@inheritdoc}
+     */
+    public function has($id)
+    {
+        return $this->getConcrete($id) != $id;
```
Comment (taylorotwell) [inline src/Illuminate/Container/Container.php]: This should probably call `->bound` instead.

**Comment by deleugpn** [comment]
@thecrypticace from the PSR

> or throw a `NotFoundExceptionInterface` if the identifier is not known to the container.

I'm interpreting this as: If you want to build something concrete, just build it yourself. The container will throw exception if the given identifier is unknown to itself.

I'm not sure if my interpretation is correct.

**Comment by thecrypticace** [comment]
Yeah, based on that, your interpretation seems sensible. 👍

**Comment by a-komarev** [comment]
Thanks!

**Pull request merged**